wagon.py

```diff
@@ -0,0 +1,1014 @@
+        zip.extractall(destination)
+
+
+def _tar(source, destination):
+    logger.info('Creating tgz archive: {0}...'.format(destination))
+    with closing(tarfile.open(destination, 'w:gz')) as tar:
+        tar.add(source, arcname=os.path.basename(source))
+
+
+def _untar(archive, destination):
+    logger.debug('Extracting tgz {0} to {1}...'.format(archive, destination))
+    with closing(tarfile.open(name=archive)) as tar:
+        tar.extractall(path=destination, members=tar.getmembers())
+
+
+def _get_wheel_tags(wheel_name):
+    filename, _ = os.path.splitext(os.path.basename(wheel_name))
+    return filename.split('-')
+
+
+def _get_platform_from_wheel_name(wheel_name):
+    """Extracts the platform of a wheel from its file name.
+    """
+    return _get_wheel_tags(wheel_name)[-1]
+
+
+def _get_platform_for_set_of_wheels(wheels_path):
+    """For any set of wheel files, extracts a single platform.
+
+    Since a set of wheels created or downloaded on one machine can only
+    be for a single platform, if any wheel in the set has a platform
+    which is not `any`, it will be used. If a platform other than
+    `any` was not found, `any` will be assumed.
+    """
+    for wheel in _get_downloaded_wheels(wheels_path):
+        platform = _get_platform_from_wheel_name(
+            os.path.join(wheels_path, wheel))
+        if platform != 'any':
+            return platform
+    return 'any'
+
+
+def _get_python_version():
+    version = sys.version_info
+    return 'py{0}{1}'.format(version[0], version[1])
+
+
+def get_platform():
+    return wheel_tags.get_platform()
+
+
+def _get_os_properties():
+    # TODO: replace with `distro`
+    return platform.linux_distribution(full_distribution_name=False)
+
+
+def _get_env_bin_path(env_path):
+    """Returns the bin path for a virtualenv
+
+    This provides a fallback for a situation in which you're trying
+    to use the script and create a virtualenv from within
+    a virtualenv in which virtualenv isn't installed and so
+    is not importable.
+    """
+    if globals().get('virtualenv'):
+        return virtualenv.path_locations(env_path)[3]
+    else:
+        return os.path.join(env_path, 'scripts' if IS_WIN else 'bin')
+
+
+def _get_default_requirement_files(source):
+    if os.path.isdir(source):
+        return [os.path.join(source, f) for f in REQUIREMENT_FILE_NAMES
+                if os.path.isfile(os.path.join(source, f))]
+
+
+def _get_pip_path(virtualenv=None):
+    pip = 'pip.exe' if IS_WIN else 'pip'
+    if virtualenv:
+        return os.path.join(_get_env_bin_path(virtualenv), pip)
+    else:
+        return os.path.join(
+            os.path.dirname(sys.executable), 'scripts' if IS_WIN else '', pip)
+
+
+def _check_installed(package, virtualenv):
+    pip_executable = _get_pip_path(virtualenv)
+    process = run('{0} freeze'.format(pip_executable), suppress_output=True)
+    if re.search(r'{0}=='.format(package), process.aggr_stdout):
```

Comment:
consider keeping a compiled pattern and using pattern.match instead of re.search